### ISSUE TITLE
IPv6 weirdness - fallback to main IPv4 IP

### ISSUE THREAD

**Issue by dazoot** [zone-eu/zone-mta#196]
We have disabled ipv6 for all our zones, but from time to time we see some weirdness in the logs.

Zone setup:
```
[mzone]
connections = 5
processes = 5
ignoreIPv6 = true
pool = "mzone"
preferIPv6 = false
```

In the log we see something like this:
```
Jul 18 11:16:21 pig zone-mta: info Sender/mzone/31862[3] 16c0427408300017f1.001 CONNECTING domain=assistenzacomputer.modena.it port=25
Jul 18 11:16:21 pig zone-mta: ERR! Sender/mzone/31862[3] 16c0427408300017f1.001 CONNFAIL domain=assistenzacomputer.modena.it local=m1.mydomain.eu[undefined] remote=2001:4b78:1001::70b:25 error=Network error when connecting to MX server m-07b.th.seeweb.it[2001:4b78:1001::70b] for assistenzacomputer.modena.it: Network is unreachable
Jul 18 11:17:25 pig zone-mta: ERR! Sender/mzone/31862[3] 16c0427408300017f1.001 CONNFAIL domain=assistenzacomputer.modena.it local=m1.mydomain.eu[undefined] remote=217.194.8.28:25 error=Network error when connecting to MX server smtp-avas.seeweb.it[217.194.8.28] for assistenzacomputer.modena.it: Connection timed out
Jul 18 11:17:25 pig zone-mta: ERR! Sender/mzone/31862[3] 16c0427408300017f1.001 CONNFAIL domain=assistenzacomputer.modena.it local=m1.mydomain.eu[undefined] remote=2001:4b78:1:20::28:25 error=Network error when connecting to MX server smtp-avas.seeweb.it[2001:4b78:1:20::28] for assistenzacomputer.modena.it: Network is unreachable
Jul 18 11:17:25 pig zone-mta: ERR! Sender/mzone/31862[3] 16c0427408300017f1.001 ERRCONNECT domain=assistenzacomputer.modena.it error=Network error when connecting to MX server m-07b.th.seeweb.it[2001:4b78:1001::70b] for assistenzacomputer.modena.it: Network is unreachable
Jul 18 11:17:25 pig zone-mta: info Sender/mzone/31862[3] 16c0427408300017f1.001 DEFERRED[recipient] from=nbounce-3119-706281-446749349-5ee16-566ab487@mydomain.eu to=redacted@assistenzacomputer.modena.it src= mx=assistenzacomputer.modena.it id=<20190718081616.27829.3119.706281.446749349.nl@mydomain.eu> (450 Network error when connecting to MX server m-07b.th.seeweb.it[2001:4b78:1001::70b] for assistenzacomputer.modena.it: Network is unreachable)
Jul 18 11:22:29 pig zone-mta: info Sender/mzone/32060[1] 16c0427408300017f1.001 DEFERRED[recipient] from=nbounce-3119-706281-446749349-5ee16-566ab487@mydomain.eu to=redacted@assistenzacomputer.modena.it src= mx=assistenzacomputer.modena.it id=<20190718081616.27829.3119.706281.446749349.nl@mydomain.eu> (450 Network error when connecting to MX server m-07b.th.seeweb.it[2001:4b78:1001::70b] for assistenzacomputer.modena.it: Network is unreachable)
Jul 18 11:27:33 pig zone-mta: info Sender/mzone/31895[4] 16c0427408300017f1.001 CONNECTING domain=assistenzacomputer.modena.it port=25
Jul 18 11:27:33 pig zone-mta: ERR! Sender/mzone/31895[4] 16c0427408300017f1.001 CONNFAIL domain=assistenzacomputer.modena.it local=m1.mydomain.eu[undefined] remote=2001:4b78:1001::70b:25 error=Network error when connecting to MX server m-07b.th.seeweb.it[2001:4b78:1001::70b] for assistenzacomputer.modena.it: Network is unreachable
Jul 18 11:27:34 pig zone-mta: info Sender/mzone/31895[4] 16c0427408300017f1.001 CONNECTED domain=assistenzacomputer.modena.it mx=smtp-avas.seeweb.it[217.194.8.28] src=m1.mydomain.eu[194.102.92.1]
Jul 18 11:27:37 pig zone-mta: info Sender/mzone/31895[4] 16c0427408300017f1.001 ACCEPTED from=nbounce-3119-706281-446749349-5ee16-566ab487@mydomain.eu to=redacted@assistenzacomputer.modena.it src=194.102.92.1 mx=smtp-avas.seeweb.it[217.194.8.28] id=<20190718081616.27829.3119.706281.446749349.nl@mydomain.eu> (250 2.0.0 Ok: queued as DC4686011D2)
Jul 18 11:27:37 pig zone-mta: info Sender/mzone/31895[4] 16c0427408300017f1.001 SMTPCLOSE [UzXK3x8RXWc] Connection closed response="250 2.0.0 Ok: queued as DC4686011D2"
```

What it seems is that there are some initial IPv4 errors, then the retry is done on IPv6 (even though we have not enabled IPv6).
After IPv6 try (can't work if we have no IPv6 ips) the mail goes out to the IPv4 remote MX with source being the server default IP: `194.102.92.1` (which is not used for mailing, not enabled in any pool).

**Comment by andris9** [comment]
The default MX for assistenzacomputer.modena.it only has a AAAA record and no A record.

How come the zone name is at first `allip_big` and then `mzone`? If you switch zones with a plugin then you may get configuration mix-ups. Using default IPv4 means that there was no IP provided by the zone to use.

**Comment by dazoot** [comment]
There was a redacted typo. `zone is always the same: mzone`.

Even if `assistenzacomputer.modena.it` has default MX with AAAA record, if IPv6 is disabled, it should not be tried at all, right ?

**Comment by dazoot** [comment]
I think the BUG is in mxconnect:
https://github.com/zone-eu/mx-connect/blob/master/lib/resolve-ip.js#L84

Here we should have:
```
if (!dnsOptions.ignoreIPv6) {
```

I have created a pull request:
https://github.com/zone-eu/mx-connect/pull/2

**Comment by andris9** [comment]
Thanks! Published as zone-mta@1.16.4
